### ISSUE TITLE
ipython error inside virtual env

### ISSUE THREAD

**Issue by dtrinf** [ipython/ipython#12791]
Ipython throws an error inside virtual env.

Is not critical for me, I only want to post it to thanks you for the great job you're doing!

Let me know if I can help you with more info.

```
  File "/home/david/python/lifebit/lib/python3.8/site-packages/traitlets/config/application.py", line 845, in launch_instance
    app.start()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/IPython/terminal/ipapp.py", line 356, in start
    self.shell.mainloop()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py", line 564, in mainloop
    self.interact()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py", line 547, in interact
    code = self.prompt_for_code()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/IPython/terminal/interactiveshell.py", line 473, in prompt_for_code
    text = self.pt_app.prompt(
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/shortcuts/prompt.py", line 1013, in prompt
    return self.app.run(set_exception_handler=set_exception_handler)
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 825, in run
    return loop.run_until_complete(
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 792, in run_async
    return await _run_async2()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 780, in _run_async2
    await self.cancel_and_wait_for_background_tasks()
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 881, in cancel_and_wait_for_background_tasks
    await task
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/buffer.py", line 1909, in new_coroutine
    await coroutine(*a, **kw)
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/buffer.py", line 1738, in async_completer
    async for completion in self.completer.get_completions_async(
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/completion/base.py", line 269, in get_completions_async
    async for completion in completer.get_completions_async(
  File "/home/david/python/lifebit/lib/python3.8/site-packages/prompt_toolkit/completion/base.py", line 196, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "/home/david/python/lifebit/lib/python3.8/site-packages/IPython/terminal/ptutils.py", line 116, in get_completions
    exc_type, exc_value, exc_tb = sys.exc_info()
NameError: name 'sys' is not defined

If you suspect this is an IPython 7.19.0 bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@python.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    %config Application.verbose_crash=True
```

**Comment by mkillianey** [comment]
See https://github.com/davidhalter/jedi/issues/1714

**Comment by MrMino** [comment]
It has been fixed in IPython **7.20** (#12793 . The fix for older versions (e.g. if you're working on Python 3.6) is to downgrade Jedi to v0.17.2:

```
pip install jedi==0.17.2
```

See if that helps. Closing this for now - if I'm mistaken and this is something else, don't hesitate to ping me.
